Source/Dafny/AST/DafnyAst.cs

```diff
@@ -3146,6 +3174,19 @@ public UserDefinedType(TypeParameter tp)
 
       return base.IsSubtypeOf(super, ignoreTypeArguments, ignoreNullity);
     }
+
+    public override bool DoesNotContainGhostConstraints() {
+      // It might not be possible to do so if the constraint of this type uses ghost predicates
+      if (this.AsSubsetType is { IsConstraintCompilable: false }) {
+        return false;
+      } else if (this.NormalizeExpandKeepConstraints() is TypeProxy) {
```

Comment:
Is this case possible? Can a `UserDefinedType` expand to a `TypeProxy` ?

Reply:
You're right, it cannot, so I'm removing this case.